### ISSUE TITLE
Cleanup `cargo hack` warnings

### ISSUE THREAD

**Issue by svaningelgem** [pola-rs/polars#10720]
### Problem description

Right now when running `cargo check` there are a LOT of warnings. This drowns out the errors.

As I'm trying to familiarize myself with the code tree, I was wondering if it's a good first step to fix all these warnings (most of them look like the imports need to have an added "cfg(feature)" check on it.

Because it's a lot of work, I prefer to first ask before even starting on it.

Thanks

**Comment by stinodego** [comment]
You should not get any warnings using the `make` commands in the `crates` folder. Basically, you'll want to run using `--all-features` mostly.

Improvements to our feature gating system are definitely welcome, but probably not very suited as a first contribution.

**Comment by svaningelgem** [comment]
What I mean is during `Test Rust` there is the check `check-features`:
```text
warning: unused imports: `timestamp_ms_to_datetime`, `timestamp_ns_to_datetime`, `timestamp_us_to_datetime`
 --> crates/polars-core/src/datatypes/any_value.rs:2:5
  |
2 |     timestamp_ms_to_datetime, timestamp_ns_to_datetime, timestamp_us_to_datetime,
  |     ^^^^^^^^^^^^^^^^^^^^^^^^  ^^^^^^^^^^^^^^^^^^^^^^^^  ^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: `#[warn(unused_imports)]` on by default

warning: unreachable pattern
   --> crates/polars-core/src/datatypes/any_value.rs:732:13
    |
732 |             dt => polars_bail!(ComputeError: "cannot get static any-value from {}", dt),
    |             ^^
    |
    = note: `#[warn(unreachable_patterns)]` on by default
```

What I would be doing is cleaning this up. The amount of logs it saves, the amount of overhead there is there... 😄 

WDYT?

**Comment by stinodego** [comment]
Right, I see what you mean. Indeed, there are many warnings when enabling just a single feature (which is what is checked in the CI for each feature with `cargo hack`).

If you can improve this _without_  littering the code with feature gates, i.e. manage to consolidate related code to a single module and feature gating the module entirely, that would be very welcome. But that is some heavy refactoring, and as I said, probably not very appropriate as a first issue.
